src/services/accounts/AccountsAssetsService.spec.ts

```diff
@@ -0,0 +1,107 @@
+import { sanitizeNumbers } from '../../sanitize/sanitizeNumbers';
+import { blockHash789629, mockApi } from '../test-helpers/mock';
+import { AccountsAssetsService } from './AccountsAssetsService';
+
+const accountsAssetsService = new AccountsAssetsService(mockApi);
+
+describe('AccountsAssetsService', () => {
+	const at = {
+		hash: '0x7b713de604a99857f6c25eacc115a4f28d2611a23d9ddff99ab0e4f1c17a8578',
+		height: '789629',
+	};
+
+	describe('AccountsAssetsService.fetchAssetBalances', () => {
+		it('Should return the correct response with the assets param', async () => {
+			const expectedResponse = {
+				at,
+				assets: [
+					{
+						assetId: '10',
+						balance: '10000000',
+						isFrozen: false,
+						isSufficient: true,
+					},
+					{
+						assetId: '20',
+						balance: '20000000',
+						isFrozen: true,
+						isSufficient: true,
+					},
+				],
+			};
+
+			const response = await accountsAssetsService.fetchAssetBalances(
+				blockHash789629,
+				'0xffff',
+				[10, 20]
+			);
+
+			expect(sanitizeNumbers(response)).toStrictEqual(expectedResponse);
+		});
+
+		it('Should return the correct response without the assets param', async () => {
+			const expectedResponse = {
+				at,
+				assets: [
+					{
+						assetId: '10',
+						balance: '10000000',
+						isFrozen: false,
+						isSufficient: true,
+					},
+					{
+						assetId: '20',
+						balance: '20000000',
+						isFrozen: true,
+						isSufficient: true,
+					},
+					{
+						assetId: '30',
+						balance: '20000000',
+						isFrozen: false,
+						isSufficient: false,
+					},
+					{
+						assetId: '30',
+						balance: '20000000',
+						isFrozen: false,
+						isSufficient: false,
+					},
+					{
+						assetId: '30',
+						balance: '20000000',
+						isFrozen: false,
+						isSufficient: false,
+					},
```

Comment:
This is a good point. In this case yes, but for practicality sake I am going to remove it so it doesnt confuse anyone because a real world scenario would have this. Instead I will make it just one.

Reply:
Probably over the top, but if in real life this shouldn't happen should we put an error check on the data from the node to make sure this is never the case?

Reply:
Yea we could do that, not sure it's totally necessary, If that were to happen, I would assume that would be a serious bug on the substrate or polkadot-js end. 

One thing to potentially filter as well though could be Assets that have a balance of zero, but that should also be up to the end user.